validation/implicit_free_surface/implicit_surface_solvers_benchmark.jl

```diff
@@ -0,0 +1,114 @@
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.BuoyancyModels: g_Earth
+using Oceananigans.ImmersedBoundaries: ImmersedBoundaryGrid, GridFittedBottom
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: FFTImplicitFreeSurfaceSolver, MGImplicitFreeSurfaceSolver
+using Printf
+
+underlying_grid = RectilinearGrid(CPU(),
+                                  topology = (Periodic, Bounded, Bounded), 
+                                  size = (64, 64, 24),
+                                  x = (-500kilometers, 500kilometers),
+                                  y = (-500kilometers, 500kilometers),
+                                  z = (-1kilometers, 0),
+                                  halo = (4, 4, 4))
+
+const Lz = underlying_grid.Lz
+const width = 50kilometers
+bump(x, y) = - Lz * (1 - 0.2 * exp(-(x^2 + y^2) / 2width^2))
+grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom(bump))
+
+# this is to fix a bug in validate_fft_implicit_solver_grid
+import Oceananigans.Models.HydrostaticFreeSurfaceModels.validate_fft_implicit_solver_grid
+validate_fft_implicit_solver_grid(ibg::ImmersedBoundaryGrid) = validate_fft_implicit_solver_grid(ibg.underlying_grid)
+
+# fft_preconditioner = FFTImplicitFreeSurfaceSolver(grid)
```

Comment:
Is the plan to keep these comments in the merge?

Reply:
I cleaned it up